.github/workflows/unit-tests.yml

```diff
@@ -121,7 +121,6 @@ jobs:
       - name: Install dependencies
         if: "!startsWith(runner.os, 'macos')"
         run: |
-          # Minimum requirements: CMake 3.9, Boost 1.69
```

Comment:
IMO this is kind of important information. We currently don't have a test for CMake 3.9 at all anymore. Hence I'd rather make that the default in the global env. Same for Boost: 1.69 is the minimum so why use another version as the default?

Reply:
thats exactly one point, we should shift the minimum up to get rid of all the workarounds. 

Stock cmake from debian bullseye is 3.18.4 - we should probably make this the new minimum

Reply:
Ubuntu 20 having CMake 3.16 is a reasonable argument to having that as the minimum as IIRC Ubuntu 18 is EOL. Maybe mention that somewhere but we shouldn't increase the required version if there isn't a benefit to not offend/inconvenience potential contributors.